pkgs/development/libraries/science/math/tensorflow-lite/default.nix

```
@@ -0,0 +1,163 @@
+    owner = "google";
+    repo = "farmhash";
+    rev = "816a4ae622e964763ca0862d9dbd19324a1eaf45";
+    sha256 = "1mqxsljq476n1hb8ilkrpb39yz3ip2hnc7rhzszz4sri8ma7qzp6";
+  };
+
+  fft2d-src = fetchurl {
+    url = "http://www.kurims.kyoto-u.ac.jp/~ooura/fft2d.tgz";
+    sha256 = "ada7e99087c4ed477bfdf11413f2ba8db8a840ba9bbf8ac94f4f3972e2a7cec9";
+  };
+
+  fp16-src = fetchFromGitHub {
+    owner = "Maratyszcza";
+    repo = "FP16";
+    rev = "4dfe081cf6bcd15db339cf2680b9281b8451eeb3";
+    sha256 = "06a8dfl3a29r93nxpp6hpywsajz5d555n3sqd3i6krybb6swnvh7";
+  };
+
+  ruy-src = fetchFromGitHub {
+    owner = "google";
+    repo = "ruy";
+    rev = "23633b37099b614a2f836ef012cafc8087fdb98c";
+    sha256 = "14k9hz6ss8qy8nsajk6lrq25f6qxrldxky31ijw0dpqnfnnswrx4";
+  };
+
+  cpuinfo-src = fetchFromGitHub {
+    owner = "pytorch";
+    repo = "cpuinfo";
+    rev = "5916273f79a21551890fd3d56fc5375a78d1598d";
+    sha256 = "0q6760xdxsg18acdv8vq3yrq7ksr7wsm8zbyan01zf2khnb6fw4x";
+  };
+
+  version = "2.5.0";
+in
+stdenv.mkDerivation {
+  pname = "tensorflow-lite";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "tensorflow";
+    repo = "tensorflow";
+    rev = "v${version}";
+    sha256 = "1jdw2i1rq06zqd6aabh7bbm0avsg4pygnfmd7gviv0blhih9054l";
+  };
+
+  buildInputs = [ zlib flatbuffers gmock ];
+
+  dontConfigure = true;
+
+  postPatch = ''
+    substituteInPlace ./tensorflow/lite/tools/make/Makefile \
+      --replace /bin/bash ${bash}/bin/bash \
+      --replace /bin/sh ${bash}/bin/sh
+  '';
+
+  buildPhase = ''
+    pushd ./tensorflow/lite/tools/make
+
+    mkdir -p ./downloads
+    pushd ./downloads
+
+    tar xzf ${fft2d-src} -C .
+
+    # enter the vendoring lair of doom
+    ln -s ${ruy-src} ./ruy
+    ln -s ${gemmlowp-src} ./gemmlowp
+    ln -s ${neon-2-sse-src} ./neon_2_sse
+    ln -s ${farmhash-src} ./farmhash
+    ln -s ${cpuinfo-src} ./cpuinfo
+    ln -s ${fp16-src} ./fp16
+    ln -s ${tflite-eigen} ./eigen
+
+    # tensorflow lite is using the *source* of flatbuffers
+    ln -s ${flatbuffers.src} ./flatbuffers
+
+    # tensorflow lite expects to compile abseil into libtensorflow-lite.a
+    ln -s ${abseil-cpp.src} ./absl
+
+    popd
+    popd
+
+    includes="\
+      -I $PWD \
+      -I $PWD/tensorflow/lite/tools/make/downloads/neon_2_sse \
+      -I $PWD/tensorflow/lite/tools/make/downloads/gemmlowp \
+      -I $PWD/tensorflow/lite/tools/make/downloads/absl \
+      -I $PWD/tensorflow/lite/tools/make/downloads/fp16/include \
+      -I $PWD/tensorflow/lite/tools/make/downloads/farmhash/src \
+      -I $PWD/tensorflow/lite/tools/make/downloads/ruy \
+      -I $PWD/tensorflow/lite/tools/make/downloads/cpuinfo \
+      -I $PWD/tensorflow/lite/tools/make/downloads/cpuinfo/src \
+      -I $PWD/tensorflow/lite/tools/make/downloads/cpuinfo/include \
+      -I $PWD/tensorflow/lite/tools/make/downloads/cpuinfo/deps/clog/include \
+      -I $PWD/tensorflow/lite/tools/make/downloads/eigen"
+
+    ${gnumake}/bin/make \
```

```suggestion
    make \
```
also make is already part of the stdenv.